doc/build-helpers/trivial-build-helpers.chapter.md

```diff
@@ -58,24 +58,108 @@ Variant of `runCommand` that forces the derivation to be built locally, it is no
 This sets [`allowSubstitutes` to `false`](https://nixos.org/nix/manual/#adv-attr-allowSubstitutes), so only use `runCommandLocal` if you are certain the user will always have a builder for the `system` of the derivation. This should be true for most trivial use cases (e.g., just copying some files to a different location or adding symlinks) because there the `system` is usually the same as `builtins.currentSystem`.
 :::
 
-## `writeTextFile`, `writeText`, `writeTextDir`, `writeScript`, `writeScriptBin` {#trivial-builder-writeText}
+## `writeTextFile`, `writeText`, `writeTextDir`, `writeScript`, `writeScriptBin`, `writeShellScript`, `writeShellScriptBin` {#trivial-builder-textwriting}
 
-These functions write `text` to the Nix store. This is useful for creating scripts from Nix expressions. `writeTextFile` takes an attribute set and expects two arguments, `name` and `text`. `name` corresponds to the name used in the Nix store path. `text` will be the contents of the file. You can also set `executable` to true to make this file have the executable bit set.
+Nixpkgs provides the following functions for producing derivations which write text into the Nix store:  `writeTextFile`, `writeText`, `writeTextDir`, `writeScript`, `writeScriptBin`, `writeShellScript`, and `writeShellScriptBin`.
 
-Many more commands wrap `writeTextFile` including `writeText`, `writeTextDir`, `writeScript`, and `writeScriptBin`. These are convenience functions over `writeTextFile`.
+`writeText`, `writeTextDir`, `writeScript`, `writeScriptBin`, `writeShellScript`, and `writeShellScriptBin` are convenience functions over `writeTextFile`.
 
-Here are a few examples:
+These are useful for creating files from Nix expressions, which may be scripts or non-executable text files.
+
+Each of these functions will cause a derivation to be realized.  When you coerce the result of each of these functions to a string, it will evaluate to the *store path* of this derivation.
```

Comment:
These functions won't cause a derivation to be realized. They will produce derivations as a side effect :sweat_smile: 

For now, they apart from creating a derivation as a side-effect, they will only produce as "direct output" an attribute set. This attribute set will contain at least the attribute `outPath` which is a special attribute as far as Nix-language is concerned.

When an attribute set `a` is string-interpolated as in `${a}`, the machinery will check if `outPath` (which precisely that symbol name) exists as an attribute on `a`: if yes, the value of `outPath` is used. 

These derivations will then later be realized ("made real?", "fully instantiated?") by other build passes. However, at this stage, we know where the artifacts of the derivation will live once made real. This is the `outPath` attribute on a `mkDerivation` style function.